mongo/datadog_checks/mongo/mongo.py

```diff
@@ -163,7 +172,13 @@ def _build_metric_list_to_collect(self):
 
     def check(self, _):
         try:
-            api = MongoApi(self._config, self.log)
+            self._check()
+        except Exception:
```

Comment:
This clause will silence the stack traces that previously got raised.

Reply:
missed ty